### ISSUE TITLE
Relay close, code: 1002

### ISSUE THREAD

**Issue by Volfor** [WalletConnect/WalletConnectFlutterV2#148]
After some time connection is closed with error code 1002, which "indicates that an endpoint is terminating the connection due to a protocol error" according to the ws protocol. Is it some relay related issue or should it be handled by the client?

**Version:**
walletconnect_flutter_v2: ^2.0.14

```
flutter: \^[[38;5;12m┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────<…>
flutter: \^[[38;5;12m│ #0   RelayClient._handleRelayClose (package:walletconnect_flutter_v2/apis/core/relay_client/relay_client.dart:293:17)<…>
flutter: \^[[38;5;12m│ #1   RelayClient._createJsonRPCProvider.<anonymous closure> (package:walletconnect_flutter_v2/apis/core/relay_client/relay_client.dart:271:11)<…>
flutter: \^[[38;5;12m├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄<…>
flutter: \^[[38;5;12m│ 💡 Handling relay close, code: 1002, reason: null<…>
flutter: \^[[38;5;12m└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────<…>
flutter: \^[[38;5;244m┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────<…>
flutter: \^[[38;5;244m│ #0   RelayClient.disconnect (package:walletconnect_flutter_v2/apis/core/relay_client/relay_client.dart:193:17)<…>
flutter: \^[[38;5;244m│ #1   RelayClient._handleRelayClose (package:walletconnect_flutter_v2/apis/core/relay_client/relay_client.dart:304:15)<…>
flutter: \^[[38;5;244m├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄<…>
flutter: \^[[38;5;244m│ RelayClient: Disconnecting from relay<…>
flutter: \^[[38;5;244m└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────<…>
````

**Comment by Volfor** [comment]
@Luzzotica it occurs after the app has been opened for a while, then connection is closed and only restart/reconnect helps

**Comment by Luzzotica** [comment]
I'll add the code to the auto-reconnect.

**Comment by Luzzotica** [comment]
Resolved with 2.0.15
